### PR TITLE
Add a known issue for TiFlash mpp.

### PR DESCRIPTION
**Pull request by windtalker**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add a known issue for TiFlash mpp.
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/use-tiflash-mpp-mode.md [hunk 1]
```diff
@@ -167,3 +167,18 @@ mysql> explain SELECT count(*) FROM test.employees;
 +------------------------------+----------+--------------+-----------------+---------------------------------------------------------+
 5 rows in set (0,00 sec)
 ```
+
+## MPP 的已知问题
+
+在 TiDB 6.5.x 及之前的版本里面，TiFlash 使用一个 query 的 `start_ts` 当做这个 query 的 unique key，在绝大多数情况下，每个 query 的 `start_ts` 都是可以唯一标识一个 query 的，但是在以下几种情况下，不同的 query 会有相同的 `start_ts`
```
Comment (ran-huang) [inline tiflash/use-tiflash-mpp-mode.md:173]:
```suggestion
在 TiDB 6.5.x 及之前的版本里面，TiFlash 使用一个查询的 `start_ts` 当做这个查询的 unique key。在绝大多数情况下，每个 query 的 `start_ts` 都可以唯一标识一个查询，但是在以下几种情况下，不同的查询会有相同的 `start_ts`：
```

File: tiflash/use-tiflash-mpp-mode.md [hunk 2]
```diff
@@ -167,3 +167,18 @@ mysql> explain SELECT count(*) FROM test.employees;
...
 5 rows in set (0,00 sec)
 ```
+
+## MPP 的已知问题
+
+在 TiDB 6.5.x 及之前的版本里面，TiFlash 使用一个查询的 `start_ts` 当做这个查询的 unique key。在绝大多数情况下，每个 query 的 `start_ts` 都可以唯一标识一个查询，但是在以下几种情况下，不同的查询会有相同的 `start_ts`：
+
+- 在同一个事务里的 query 都有相同的 `start_ts`
+- 用 [`tidb_snapshot`](/system-variables.md#tidb_snapshot) 来指定读取特定历史时刻数据的时候
+- 开启了 [Stale Read](/stale-read.md) 的时候
```
Comment (ran-huang) [inline tiflash/use-tiflash-mpp-mode.md:177]: 这两种情况下，哪些查询会有相同的 `start_ts` 呢？

File: tiflash/use-tiflash-mpp-mode.md [hunk 3]
```diff
@@ -167,3 +167,18 @@ mysql> explain SELECT count(*) FROM test.employees;
...
+
+当 `start_ts` 无法唯一表示 MPP query 的时候，如果 TiFlash 在同一时刻看到不同的 query 拥有相同的 `start_ts` 时，就可能会报错。典型的报错情况如下：
+
+- 当多个相同的 `start_ts` 的 query 同时发给 TiFlash 时，可能会遇到 "task has been registered" 的报错
+- 当同一个事务里连续跑一些简单的带 limit 的 query 时，因为 TiDB 在 limit 条件满足后会给 TiFlash 发送 cancel request 的请求，这个请求也是以 `start_ts` 来标识需要 cancel 的 query 的，假如 TiFlash 中有其他相同 `start_ts` 的 query 的话，那其他 query 就可能会被误 cancel。例如[这个 issue](https://github.com/pingcap/tidb/issues/43426) 里面碰到的问题
```
Comment (ran-huang) [inline tiflash/use-tiflash-mpp-mode.md:182]:
```suggestion
- 当同一个事务里连续执行多个简单的使用 `LIMIT` 的查询时，TiDB 在 limit 条件满足后会给 TiFlash 发送 cancel request 的请求，这个请求也以 `start_ts` 来标识需要 cancel 的查询。假如 TiFlash 中有其他相同 `start_ts` 的查询的话，那其他查询就可能会被误 cancel。例如[这个 issue](https://github.com/pingcap/tidb/issues/43426) 里面碰到的问题。
```

File: tiflash/use-tiflash-mpp-mode.md [hunk 4]
```diff
@@ -167,3 +167,18 @@ mysql> explain SELECT count(*) FROM test.employees;
...
+
+该问题已在 TiDB 6.6.0 中 fix
```
Comment (ran-huang) [inline tiflash/use-tiflash-mpp-mode.md:184]:
```suggestion
该问题已在 TiDB v6.6.0 中修复，建议使用[最新的 LTS 版本](https://docs.pingcap.com/zh/tidb/stable)。
```

File: tiflash/use-tiflash-mpp-mode.md (hunk 2 above)
Comment (windtalker) [reply tiflash/use-tiflash-mpp-mode.md:177]: 比如 snapshot 设置的一样，那他的start_ts就是一样的

Comment (ran-huang) [inline tiflash/use-tiflash-mpp-mode.md:177]: @windtalker 这样描述可以吗
```suggestion
- 用 [`tidb_snapshot`](/system-variables.md#tidb_snapshot) 来指定读取特定历史时刻数据时，手动指定了相同的时间点
- 开启了 [Stale Read](/stale-read.md) 时，手动指定了相同的时间点
```

Comment (windtalker) [reply tiflash/use-tiflash-mpp-mode.md:177]: 可以的

File: tiflash/use-tiflash-mpp-mode.md [hunk 5]
```diff
@@ -167,3 +167,18 @@ mysql> explain SELECT count(*) FROM test.employees;
 +------------------------------+----------+--------------+-----------------+---------------------------------------------------------+
 5 rows in set (0,00 sec)
 ```
+
+## MPP 的已知问题
+
+在 TiDB 6.5.x 及之前的版本里面，TiFlash 使用一个查询的 `start_ts` 当做这个查询的 unique key。在绝大多数情况下，每个 query 的 `start_ts` 都可以唯一标识一个查询，但是在以下几种情况下，不同的查询会有相同的 `start_ts`：
```
Comment (ran-huang) [inline tiflash/use-tiflash-mpp-mode.md:173]:
```suggestion
在当前版本中，TiFlash 使用一个查询的 `start_ts` 当做这个查询的 unique key。在绝大多数情况下，每个 query 的 `start_ts` 都可以唯一标识一个查询，但是在以下几种情况下，不同的查询会有相同的 `start_ts`：
```

修改理由：避免 cp 之后还需要改

**Review by ran-huang — APPROVED** [review]
rest LGTM

**Pull request merged**